libcudacxx/include/cuda/std/__type_traits/type_list.h

```diff
@@ -58,6 +67,49 @@ struct __type_quote
   using __apply = _Fn<_Ts...>;
 };
 
+//! @brief Turns a unary class or alias template into a meta-callable
+template <template <class> class _Fn>
+struct __type_quote1
+{
+  template <class _Ty>
+  using __apply = _Fn<_Ty>;
+};
```

Comment:
Why don't we have just a variadic quote?
```suggestion
template <template <class> class _Fn>
struct __type_quote
{
  template <class... _Ty>
  using __apply = _Fn<_Ty...>;
};
```

Reply:
There already is one. It's on line 64.

Reply:
Oh sorry, I was just looking around @miscco's comment. Then why do we need `__type_quote1` and `__type_quote2` at all?

Reply:
because expanding a pack into a unary template will fail in certain situations, even if the pack has exactly one type in it.

Reply:
ok, now I am curious (sorry :D). do you have an example for me? I have never seen this!

Reply:
i just ran into an instance of this problem while trying to optimize `__type_fold_right`. this code is failing while trying to expand a pack of one element into a template with one parameter. https://godbolt.org/z/EK1ofe4G1